tests/test_doctor.py

```
@@ -118,6 +118,129 @@ def test_open_readonly_doctor_db_rejects_schema_and_data_writes(tmp_path):
+    monkeypatch.setitem(sys.modules, "sqlite_vec", fake_sqlite_vec)
+
+    assert doctor._load_optional_sqlite_vec(cast(sqlite3.Connection, TrackingConnection())) is True
+    assert calls == [True, "load", False]
+
+
+def test_optional_sqlite_vec_load_fails_closed_when_disable_fails(monkeypatch):
+    calls: list[object] = []
+    fake_sqlite_vec = types.SimpleNamespace(load=lambda _conn: calls.append("load"))
+
+    class DisableFailingConnection:
+        def enable_load_extension(self, enabled):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `enable_load_extension`

Add return type annotation: `None`

(ANN202)